src/libs/ReportUtils.js

```diff
@@ -2667,6 +2669,55 @@ function buildOptimisticApprovedReportAction(amount, currency, expenseReportID)
     };
 }
 
+/**
+ * Builds an optimistic MOVED report action with a randomly generated reportActionID.
+ * This action is used when we move reports across workspaces.
+ *
+ * @param {String} fromPolicyID
+ * @param {String} toPolicyID
+ * @param {Number} newParentReportID
+ * @param {Number} movedReportID        An ID of the report we are moveing across workspaces
+ *
+ * @returns {Object}
+ */
+function buildOptimisticMovedReportAction(fromPolicyID, toPolicyID, newParentReportID, movedReportID) {
+    const originalMessage = {
+        fromPolicyID,
+        toPolicyID,
+        newParentReportID,
+        movedReportID,
+    };
+
+    const policyName = getPolicyName(allReports[`${ONYXKEYS.COLLECTION.REPORT}${newParentReportID}`]);
+
+    return {
+        actionName: CONST.REPORT.ACTIONS.TYPE.MOVED,
+        actorAccountID: currentUserAccountID,
+        automatic: false,
+        avatar: lodashGet(currentUserPersonalDetails, 'avatar', UserUtils.getDefaultAvatarURL(currentUserAccountID)),
+        isAttachment: false,
+        originalMessage,
+        message: [
+            {
+                html: `moved the report to the <a href='${CONST.NEW_EXPENSIFY_URL}r/${newParentReportID}' target='_blank' rel='noreferrer noopener'>${policyName}</a> workspace`,
+                text: `moved the report to the ${policyName} workspace`,
```

Comment:
For v1 they are fine to be in English, its a new action type to be handled in NewDot and we would make these polish adjustments in v2. We have been collecting the improvements in the tracking issue https://github.com/Expensify/App/issues/31314 which is added to the wave9 project for once this will be worked on